native/addon.cpp

```diff
@@ -77,8 +77,9 @@ namespace {
 
 	template <size_t index, typename I, typename A>
 	unsigned int assumeCompression (const I& info, const A& p) {
-		if (info.Length() <= index) return __isPointCompressed(p) ? SECP256K1_EC_COMPRESSED : SECP256K1_EC_UNCOMPRESSED;
-		if (info[index]->IsUndefined()) return SECP256K1_EC_COMPRESSED;
+		if (info.Length() <= index || info[index]->IsUndefined()) {
+			return __isPointCompressed(p) ? SECP256K1_EC_COMPRESSED : SECP256K1_EC_UNCOMPRESSED;
+		}
```

Comment:
Current behavior when uncompressed key is passed in:

||native|JS|
|---|---|---|
|no compressed arg (implicit undefined)|UC|UC|
|pass undefined to compressed arg|C|UC|

This solution changes it to 

||native|JS|
|---|---|---|
|no compressed arg (implicit undefined)|UC|UC|
|pass undefined to compressed arg|UC|UC|

Also note that explicitly passing falsey values like 0, '', null, and others, will be treated the same as `false`

Reply:
Note that this discrepancy in JS and Native exists by the nature of assumeCompression not having access to all the arguments of the calling function...

Another way to remove the discrepancy would be to pass `arguments` as an arg to assumeCompression similar to the Native code...

Perhaps JS should adhere to Native here... but I feel like the JS logic makes more sense. "If undefined, follow the key's compressed attribute" should be the same regardless of implicit undefined (no arg) and explicit undefined.

Reply:
Which is why I solved it this way.

Reply:
Since the only people affected by this change would be people who pass in an uncompressed pubkey along with explicitly passing undefined for compressed. (Though notably, that could also be people who wanted to pass true but misspelled the object parameter. If they were using native they wouldn't have noticed their typo thanks to our behavior)

Reply:
I don't think a major semver bump is needed.